CRISPResso2/CRISPRessoCORE.py

```diff
@@ -501,6 +862,488 @@ def custom_encoder(obj):
+                                        READ_LENGTH = len(variant[match_name]['aln_seq'])
+                                    N_GLOBAL_SUBS += (variant[match_name]['substitution_n'] + variant[match_name]['substitutions_outside_window'])
+                                    N_SUBS_OUTSIDE_WINDOW += variant[match_name]['substitutions_outside_window']
+                                    N_MODS_IN_WINDOW += variant[match_name]['mods_in_window']
+                                    N_MODS_OUTSIDE_WINDOW += variant[match_name]['mods_outside_window']
+                                    if variant[match_name]['irregular_ends']:
+                                        N_READS_IRREGULAR_ENDS += 1
+                        else:
+                            N_TOT_READS += 1
+                            N_COMPUTED_ALN += 1
+                            variantCache[variant[match_name]['aln_seq']] = variant
+                            if len(variant['aln_ref_names']) == 1 or args.expand_ambiguous_alignments:                   
+                                for name in variant['aln_ref_names']:
+                                    match_name = "variant_" + name
+                                    if READ_LENGTH == 0:
+                                        READ_LENGTH = len(variant[match_name]['aln_seq'])
+                                    N_GLOBAL_SUBS += (variant[match_name]['substitution_n'] + variant[match_name]['substitutions_outside_window'])
+                                    N_SUBS_OUTSIDE_WINDOW += variant[match_name]['substitutions_outside_window']
+                                    N_MODS_IN_WINDOW += variant[match_name]['mods_in_window']
+                                    N_MODS_OUTSIDE_WINDOW += variant[match_name]['mods_outside_window']
+                                    if variant[match_name]['irregular_ends']:
+                                        N_READS_IRREGULAR_ENDS += 1
+
+                fastq1_id = fastq1_file.readline()
+                fastq2_id = fastq2_file.readline()
+
+        if fastq1_id is not None or fastq2_id is not None:
+            error("The two fastq files are not the same length. Please check your input files.")
+
+        fastq1_file.close()
+        fastq2_file.close()
+
+    else:
+        if fastq1_filename.endswith('.gz'):
+            fastq1_handle = gzip.open(fastq1_filename, 'rt')
+        else:
+            fastq1_handle=open(fastq1_filename)
+
+        if fastq2_filename.endswith('.gz'):
+            fastq2_handle = gzip.open(fastq2_filename, 'rt')
+        else:
+            fastq2_handle=open(fastq2_filename)
+
+        if args.fastq_output:
+            fastq_write_out_handle = open(fastq_write_out_file, 'w')
+
+        not_aln = {} #cache for reads that don't align
+        while(True):
+            #read through fastq in sets of 4
+            fastq1_id = fastq1_handle.readline().strip()
+            if not fastq1_id:
+                break
```

Comment:
It seems like the case where r1 and r2 don't have the same number of reads isn't handled in this single threaded version. I think that `fastq2_id = fastq2_handle.readline().strip()` should be moved up here and then this can be refactored to something like this:

```
if not fastq1_id and not fastq2_id:
    break
elif (fastq1_id is None and fastq2_id is not None) or (fastq1_id is not None and fastq2_id is None):
    error("The two fastq files are not the same length. Please check your input files.")
```